### PR TITLE
Fixes LX and SlimyTreeTaps rubber conflict

### PR DESCRIPTION
**Pull request by NCBPFluffyBear**
## Short Description
LX Rubber no longer registers if SlimyTreeTaps is on the server.

## Additions/Changes/Removals
Moved rubber registration into its method that checks if SlimyTreeTaps is enabled

## Related Issues
#45 

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with base Slimefun and made sure nothing breaks/unexpected happens.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [x] I did my best to add documentation to any public classes or methods I added which may not be obvious to maintainers.
- [x] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values

### CODE REVIEW

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java [hunk 1]
```diff
@@ -48,17 +53,21 @@ private void registerTools() {
...
-        new RubberSynthesizer().register(LiteXpansion.getInstance());
         new RefinedSmeltery().register(LiteXpansion.getInstance());
     }
 
-    private void registerMiscItems() {
-        //Rubber
-        registerNonPlaceableItem(Items.RUBBER, RubberSynthesizer.RECIPE_TYPE, SlimefunItems.OIL_BUCKET);
+    //Disable when SlimyTreeTaps exists
+    private void registerRubber() {
+        if (!getServer().getPluginManager().isPluginEnabled("SlimyTreeTaps")) {
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]: Should rather check "isPluginEnabled()" instead of a null check here. xx cookie

**Review by J3fftw1 — CHANGES_REQUESTED** [review]
you also need to soft depend it in the plugin.yml

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java [hunk 2]
```diff
@@ -21,12 +21,16 @@
 
 import javax.annotation.Nonnull;
 
+import static org.bukkit.Bukkit.getServer;
```
Comment (WalshyDev) [inline src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]: No

**Review by WalshyDev — CHANGES_REQUESTED** [review]

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java (hunk 2 above)
Comment (NCBPFluffyBear) [reply src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]: How do I fix this?

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java (hunk 1 above)
Comment (NCBPFluffyBear) [reply src/main/java/dev/j3fftw/litexpansion/ItemSetup.java:63]: > you also need to soft depend it in the plugin.yml

Its already soft depended, you guys added it a while ago.

**Comment by NCBPFluffyBear** [comment]
> you also need to soft depend it in the plugin.yml

Its already soft depended, you guys added it a while ago.

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java (hunk 2 above)
Comment (WalshyDev) [reply src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]: Once again, use the actual Class.method() rather than doing a stupid static import.

**Review by WalshyDev — APPROVED** [review]
LGTM

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java [hunk 3]
```diff
@@ -27,6 +28,8 @@
     private final ItemStack glass = new ItemStack(Material.GLASS);
     private boolean initialised;
 
+    private final ItemStack rubberItem = SlimefunItem.getByID("RUBBER").getItem();
```
Comment (TheBusyBiscuit) [inline src/main/java/dev/j3fftw/litexpansion/ItemSetup.java:31]: This will throw a NPE. Like... everytime when TreeTaps is **not** installed.

**Pull request merged**